hypothesis-python/tests/quality/test_float_shrinking.py

```diff
@@ -45,3 +45,23 @@ def test_shrinks_downwards_to_integers_when_fractional(b):
         ).filter(lambda x: int(x) != x)
     )
     assert g == b + 0.5
+
+
+@pytest.mark.parametrize("s", range(10))
+def test_shrinks_to_canonical_nan(s):
```

Comment:
I regret adding this test, as it is very unreliable - minor changes such as `lambda a: a != a` → `math.isnan` makes it not find the problem. Do we have a simple way to explicitly specify the starting point of a shrink?

Reply:
(and such failure to find the problem will never be recognized of course, since the problem itself is fixed)

Reply:
yup, tests.common.shrinking_from, see eg `test_zero_examples_with_variable_min_size`